### ISSUE TITLE
Offline install bundle external 

### ISSUE THREAD

**Issue by Nova38** [twpayne/chezmoi#4090]
## Is your feature request related to a problem? Please describe.

When installing dot files in a offline or airgaped environment getting external content is difficult and requires either using the archive command and then manually installing the dot files or not using the chezmoiexternal feature. 

## Describe the solution you'd like

I would like to propose a option to download the externals to an archive that could then be transferred along with the local copy of the dotfiles git repo to the offline/airgaped machine. Then when specifying the location of got repo in the unit command add a flag to also tell it where to find the externals archive. It would then copy the files out of the archive to the locations specified in the 
chezmoiexternal files. Optionally allow specifying the machine specs/data so as to anwser templated files.

## Describe alternatives you've considered

- The closest solution in the archive command. However this loses the ability to template files. 
- Just not use the externals and manually add the files to the dotfiles repo, which would make update it more challenging.
- alternatively add the files to a ignore file and use a custom bootstrap script to pre download them.

## Additional context

An offline environment could include use cases such as a vm, or during presseeding/autoinstall where the repo could be stored on local media, a git server, or network storage.

An example of something similar would be how terraform provideds a way to mirror the providers to a local folder that can be copied down.

https://support.hashicorp.com/hc/en-us/articles/23562100651923-How-to-use-Terraform-CLI-with-Local-Mirror-for-Provider-Plugins-for-system-without-internet-access

**Comment by KapJI** [comment]
This can be solved by adding  support for `file://` schema proposed in https://github.com/twpayne/chezmoi/issues/4072.

**Comment by Nova38** [comment]
Ah I didn't see that proposal. The proposed `file://` would potentially resolve the issue. From my reading of that issue  it would still require a decent about of manual labor to initially cache the needed files and then update the URL to include the file saved location. It would also require changing every external to have multiple urls and require that the end user comes up with a structure for the archive. 

I think that there definitely is some overlap between the two proposals though. If I read that issue correctly the `file://` proposals seeks to allow the end user to provide a copy instead of letting chezmoi download and cache it. This proposal seeks to have chezmoi preload the cache and then transfer it over and transparently load them.  Though the implementation of the two proposals would probably have a bit of an overlap.

**Comment by twpayne** [comment]
You can run [`chezmoi archive`](

**Comment by twpayne** [comment]
Note you can still use templating: use a different config file when generating the archive with the appropriate variables set.

**Comment by twpayne** [comment]
> I would like to propose a option to download the externals to an archive that could then be transferred along with the local copy of the dotfiles git repo to the offline/airgaped machine.

Hopefully #4112 should fix this.

tl;dr:

* You can now specify multiple URLs for finding externals by adding `urls` property to your externals.
* chezmoi uses the first external URL that succeeds, so you can use a `file://` URL that points to an archive that you copy along with your source directory.

@Nova38 would you be able to test this?

**Comment by Nova38** [comment]
> > I would like to propose a option to download the externals to an archive that could then be transferred along with the local copy of the dotfiles git repo to the offline/airgaped machine.
> 
> Hopefully #4112 should fix this.
> 
> tl;dr:
> 
> * You can now specify multiple URLs for finding externals by adding `urls` property to your externals.
> * chezmoi uses the first external URL that succeeds, so you can use a `file://` URL that points to an archive that you copy along with your source directory.
> 
> @Nova38 would you be able to test this?

That sounds like a great solution, I can test it first thing tomorrow!

**Comment by twpayne** [comment]
Thanks, but please no rush. I'm thinking of doing the next release of chezmoi sometime in the next couple of weeks, so please take the time that you need.

**Comment by twpayne** [comment]
This was released in chezmoi v2.55.0.
